lib/full_text_search/similar_searcher/mroonga.rb

```diff
@@ -11,20 +11,32 @@ def self.included(base)
 
       module InstanceMethods
         def similar_issues(user: User.current, project_ids: [], limit: 5)
-          desc = [subject, description, journals.sort_by(&:id).map(&:notes)].flatten.join("\n")
+          # NOTE: The sanitize_sql_array method in the MySQL adapter is not
+          # schema-aware. It quotes numeric parameters as strings to prevent
+          # query manipulation attacks. However, if numeric parameters like :id
+          # and :limit are not explicitly converted to strings beforehand, this
+          # can lead to a syntax error.
+          #
+          # For example, without the explicit conversion, the following error is
+          # caused
+          #
+          #   ...'limit', ''5'', 'sor' at line 5
+          #
+          # To prevent such syntax errors, we explicitly convert numeric
+          # parameters to strings before passing them to sanitize_sql_array.
           sql = <<-SQL.strip_heredoc
           select mroonga_command(
                    'select',
                    'table', 'issue_contents',
                    'output_columns', 'issue_id, _score',
-                   'filter', CONCAT('(content *S "', mroonga_escape(:desc), '") && issue_id != :id', ' && #{filter_condition(user, project_ids)}'),
-                   'limit', ':limit',
+                   'filter', CONCAT('(content *S "', mroonga_escape(:term), '") && issue_id != ', :id, ' && #{filter_condition(user, project_ids)}'),
```

Comment:
`content` is better than `term` because we use normal sentences in similar search not like boolean full text search. Boolean full text search uses keywords. In boolean full text search context, `term` isn't strange.

Reply:
fix: [bb02a38](https://github.com/clear-code/redmine_full_text_search/pull/167/commits/bb02a3830f435063504293ed4413668f9b5b27b3) Thanks, I think I misunderstood the similar search. I thought it used the keywords like boolean full text search.